src/main/scala/NumberToText.scala

```diff
@@ -0,0 +1,23 @@
+import scala.annotation.tailrec
+
+class NumberToText(number : BigInt) {
+  def spliceLongNum(): List[String] = {
+    @tailrec def spliceRec(remainSegments : List[Char], splicedNums : List[String]) : List[String] = remainSegments.length match {
+      case x if x > 3 => spliceRec(remainSegments.dropRight(3), splicedNums :+ remainSegments.takeRight(3).mkString )
```

Comment:
.takeRight(3).mkString <-- is getting called quite a few times, could be a helper function that does this for you